app/components/search.tsx

```diff
@@ -378,3 +379,41 @@ const normalize = (question: string) =>
  * Sort function for the highest score on top
  */
 const byScore = (a: SearchResult, b: SearchResult) => b.score - a.score
+
+/**
+ * Handle flushing searched phrases to the NLP logger endpoint
+ */
+let prevSearch = ''
+let lastTimestamp = 0
```

Comment:
I very much dislike these variables here, but I couldn't come up with a better way of keeping state. It's fine from a concurrency point of view, as the `logSearch` function should be called sequentially, anyway